src/Uno.UI/DataBinding/BindingExpression.cs

```diff
@@ -309,16 +301,17 @@ internal void RefreshTarget()
 		}
 
 		/// <summary>
-		/// Turns UpdateSourceTrigger.Default to DependencyProperty's FrameworkPropertyMetadata.DefaultUpdateSourceTrigger
+		/// Turns UpdateSourceTrigger.Default to PropertyChanged, except for TextBox.TextProperty it's Explicit
 		/// </summary>
-		/// <returns></returns>
+		/// <remarks>
+		/// For TextBox.TextProperty, it should be LostFocus, but for now, it's Explicit and we are getting the
+		/// same behavior by explicitly updating the binding on losing focus in TextBox code.
+		/// </remarks>
 		private UpdateSourceTrigger ResolveUpdateSourceTrigger()
 		{
 			if (ParentBinding.UpdateSourceTrigger == UpdateSourceTrigger.Default)
 			{
-				var metadata = TargetPropertyDetails.Property?.GetMetadata(_targetOwnerType) as FrameworkPropertyMetadata;
-				return metadata?.DefaultUpdateSourceTrigger
-					?? UpdateSourceTrigger.PropertyChanged;
+				return TargetPropertyDetails.Property == TextBox.TextProperty ? UpdateSourceTrigger.Explicit : UpdateSourceTrigger.PropertyChanged;
```

Comment:
GetMetadata can commonly be a dictionary lookup. We avoid unnecessary dictionary lookup now